File: src/main/java/io/playground/orderservice/infrastructure/worker/outbox/OutboxWorkerProperties.java
package io.playground.orderservice.infrastructure.worker.outbox;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "tuning.outbox-worker")
@Getter
@AllArgsConstructor
public class OutboxWorkerProperties {
    private final long schedulingInterval;
    private final int poolSize;
    private final int limitSize;
    private final int retryMax;
}

File: src/main/kotlin/io/playground/orderservice/infrastructure/worker/outbox/OutboxWorkerProperties.kt
package io.playground.orderservice.infrastructure.worker.outbox

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "tuning.outbox-worker")
data class OutboxWorkerProperties(
    val schedulingInterval: Long,
    val poolSize: Int,
    val limitSize: Int,
    val retryMax: Int,
)
